### PR TITLE
NF: support to export NIFTI to CoSMo format

### PR DESCRIPTION
**Pull request by nno**
WiP for #430 (@mvdoc @snastase). This first step allows exporting an fMRI dataset created in PyMVPA to CoSMoMVPA.

TODO for this PR: update CoSMoMVPA's ``cosmo_fmri_dataset`` so that the resulting struct can be read in CoSMoMVPA directly.

Roadmap: 
- implement i/o for the reverse way, i.e. allow an fmri dataset from CoSMoMVPA to be imported in PyMVPA, and be recognized as nifti-like frmi-dataset
- to do same for surface-based data (bi-directional)

### CODE REVIEW

**Comment by nno** [comment]
The CoSMoMVPA-part has been merged into master. @yarikoptic @hanke, would you be available for a quick review?

File: mvpa2/datasets/cosmo.py [hunk 1]
```diff
@@ -369,24 +372,28 @@ def _mat_replace_matlab_function_by_string(x):
...
     Notes
     -----
-    scipy can read but not write Matlab function hanndles; the use case of
+    scipy can read but not write Matlab function handles; the use case of
     this function is to replace such function handles by something that scipy
     can write
     '''
 
-    if isinstance(x, matlab.mio5_params.MatlabFunction):
+    unsupported_classes=(matlab.mio5_params.MatlabFunction,
```
Comment (yarikoptic) [inline mvpa2/datasets/cosmo.py]: Pep8 spaces?
BTW I would recommend pycharm for python development ;-)

Comment (nno) [inline mvpa2/datasets/cosmo.py]: I *am* using pycharm, and used its auto-format code option...

File: mvpa2/tests/test_cosmo.py [hunk 2]
```diff
@@ -326,3 +359,34 @@ def test_cosmo_repr_and_str():
...
+        p_value = pymvpa_ds.a[k].value
+
+        if isinstance(p_value, tuple):
+            c_value = c_value.ravel()
+            p_value = np.asarray(p_value).ravel()
+
+        assert_array_almost_equal(c_value, p_value)
+
+    fa_dict=dict(_obj2tup(cosmomvpa_struct['fa']))
+    fa_keys=['voxel_indices']
```
Comment (yarikoptic) [inline mvpa2/tests/test_cosmo.py]: Pep 8 spaces

File: mvpa2/datasets/cosmo.py (hunk 1 above)
Comment (yarikoptic) [inline mvpa2/datasets/cosmo.py]: Ah cool
It must have warned about spaces here

**Comment by yarikoptic** [comment]
If it works, then I guess everything is kosher. Eventually we should get some kind of integration test with full round trip to Cosmo and back

**Comment by nno** [comment]
@yarikoptic It seems I did not run PyCharm auto-format on the latest code - it should be updated now, following PEP8 better.

**Comment by yarikoptic** [comment]
Yeah since empty lines are gone! That is a fix on its own.
So do you want us to merge it now, out you want to get those to-do items into this pr before then?

**Comment by nno** [comment]
@yarikoptic, thanks for the review. I guess we can merge now, and then do the other parts in a separate PR - that reduces the workload for the further pieces.

**Comment by yarikoptic** [comment]
OK feel free to merge our I will do whether Travis passes again

**Comment by nno** [comment]
@yarikoptic Thanks, there's no rush so let's just wait until Travis is happy

**Comment by nno** [comment]
Thanks @yarikoptic.

**Pull request merged**